src/github/auto-merge.ts

```diff
@@ -69,16 +64,46 @@ export class AutoMerge extends Component {
 
     const approvedReviews = options.approvedReviews ?? 1;
 
+    // add initial conditions (additional conditions can be added later)
+    this.addConditions(`#approved-reviews-by>=${approvedReviews}`);
+    this.addConditions(...blockingCondition);
+
     mergify.addRule({
       name: 'Automatic merge on approval and successful build',
       actions: mergeAction,
-      conditions: [
-        `#approved-reviews-by>=${approvedReviews}`,
-        ...blockingCondition,
-        ...successfulBuild,
-      ],
+      conditions: (() => this.renderConditions()) as any,
     });
 
     this.project.addPackageIgnore('/.mergify.yml');
   }
+
+  /**
+   * Adds conditions to the auto merge rule.
+   * @param conditions The conditions to add (mergify syntax)
+   */
+  public addConditions(...conditions: string[]) {
+    this.conditions.push(...conditions);
```

Comment:
Seems like we don't really need the `conditions` array if we already have the `lazyConditions`.

```suggestion
    this.lazyConditions.push({ renderConditions: () => conditions });
```

Reply:
Done.